src/Script.sol

```diff
@@ -11,4 +11,50 @@ abstract contract Script {
         address(bytes20(uint160(uint256(keccak256('hevm cheat code')))));
 
     Vm public constant vm = Vm(VM_ADDRESS);
+
+    // Calcuate the corresponding contract creation address for a given address and nonce
+    function addressFrom(address origin, uint256 nonce) internal pure returns (address creation) {
+        bytes memory data;
+        if (nonce == 0x00)
+            data = abi.encodePacked(bytes1(0xd6), bytes1(0x94), origin, bytes1(0x80));
+        else if (nonce <= 0x7f)
+            data = abi.encodePacked(bytes1(0xd6), bytes1(0x94), origin, uint8(nonce));
+        else if (nonce <= 0xff)
+            data = abi.encodePacked(
+                bytes1(0xd7),
+                bytes1(0x94),
+                origin,
+                bytes1(0x81),
+                uint8(nonce)
+            );
+        else if (nonce <= 0xffff)
+            data = abi.encodePacked(
+                bytes1(0xd8),
+                bytes1(0x94),
+                origin,
+                bytes1(0x82),
+                uint16(nonce)
+            );
+        else if (nonce <= 0xffffff)
+            data = abi.encodePacked(
+                bytes1(0xd9),
+                bytes1(0x94),
+                origin,
+                bytes1(0x83),
+                uint24(nonce)
+            );
+        else
+            data = abi.encodePacked(
+                bytes1(0xda),
+                bytes1(0x94),
+                origin,
+                bytes1(0x84),
+                uint32(nonce)
+            );
+        bytes32 hash = keccak256(data);
+        assembly {
+            mstore(0, hash)
+            creation := mload(0)
```

Comment:
why do you mstore then mload here? also memslot 0 is unstable so if you want it around, use address at free mem pointer and return it. but more likely feels like it should just be removed?

Reply:
I think it was to avoid all the casting necessary to go from bytes32 to address but obviously the casting is neater / more explicit. It's very old code and it's been replaced by the solmate impl in my latest commit anyway as recommended by @transmissions11.